Realm/RLMNetworkTransport.mm

```diff
@@ -156,7 +156,8 @@ - (void)URLSession:(__unused NSURLSession *)session
     response.httpStatusCode = httpResponse.statusCode;
 
     if (error) {
-        response.body = [error localizedDescription];
+        response.body = error.localizedDescription;
+        response.customStatusCode = error.code;
```

Comment:
Would it make sense to have a test that ensures the `customStatusCode` gets mapped to the expected code?

Reply:
Main thing here is to expose a code and any error will have one.